### DISCUSSION TITLE
Removing the 65,000 rules limit in speedy mode due to IE being deprecated

### DISCUSSION THREAD

**Discussion by poteboy** [emotion-js/emotion#3018] [Ideas]
As Internet Explorer has been deprecated and is no longer supported by Microsoft, I would like to discuss the possibility of reevaluating the 65,000 rules limit in speedy mode for Emotion.

I assume that the current limit of 65,000 rules might be based on the maximum number of rules allowed in a single stylesheet for IE 10. By keeping this limit, we might be unnecessarily limiting the capabilities of Emotion for users who do not use Internet Explorer.

I suggest that we consider updating Emotion to remove or increase the limit of rules in speedy mode to better accommodate modern browsers' capabilities. This would provide an improved experience for developers using Emotion in browsers with higher or no specific limits.

I would appreciate any thoughts or feedback on this matter. If there is a consensus that removing or increasing the limit is a reasonable change, I would be happy to submit a pull request for this update.

**Comment by Andarist** [comment]
How the current experience is being affected by this limit though?

**Reply by poteboy** [reply]
Thank you for your response. I understand that the current limit might not have a significant impact on most projects, and revisiting the limit might not be a high priority.

My intention in raising this discussion was to explore the possibility of optimizing Emotion for modern browsers and to learn more about the rationale behind the current limit. I appreciate the insights you have provided.

If there isn't a strong need to remove or increase the limit, and the current implementation works well for the majority of users, I agree that we can consider this discussion closed. Thank you for your time and consideration.

**Reply by Andarist** [reply]
I'm actually not sure what was the reason behind this limit, maybe it originally was because of IE. It's quite harmless though so I don't see a strong reason to change it today without further analysis and that would cost some hours (to recheck all popular browsers and all supported versions)

**Comment by poteboy** [comment]
I understand that reevaluating the limit would require significant time and effort, and since there is no strong reason to change it at this moment, I agree that it's reasonable to keep the limit as it is for now. I appreciate your input on the matter, and if any issues related to the limit arise in the future, we can always revisit the topic. Thanks again for your time!
